### ISSUE TITLE
tiff fails to install on Linux: `configure: error: libdeflate library not found at /opt/local/lib`

### ISSUE THREAD

**Issue by barracuda156** [RJVB/macstrop#118]
```
--->  tiff-4.6.0.tar.gz does not exist in /opt/local/var/macports/distfiles/tiff
--->  Attempting to fetch tiff-4.6.0.tar.gz from https://download.osgeo.org/libtiff/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 3500k  100 3500k    0     0   767k      0  0:00:04  0:00:04 --:--:--  784k
--->  Verifying checksums for tiff
--->  Checksumming tiff-4.6.0.tar.gz
--->  Extracting tiff
--->  Extracting tiff-4.6.0.tar.gz
--->  Configuring tiff
configure: WARNING: unrecognized options: --with-x
checking build system type... riscv64-unknown-linux-gnu
checking host system type... riscv64-unknown-linux-gnu
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether UID '0' is supported by ustar format... yes
checking whether GID '0' is supported by ustar format... yes
checking how to create a ustar tar archive... gnutar
checking whether to enable maintainer-specific portions of Makefiles... no
checking whether make supports the include directive... yes (GNU style)
checking for gcc... /usr/bin/clang
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether /usr/bin/clang accepts -g... yes
checking for /usr/bin/clang option to accept ISO C89... none needed
checking whether /usr/bin/clang understands -c and -o together... yes
checking dependency style of /usr/bin/clang... gcc3
checking for /usr/bin/clang option to accept ISO C99... none needed
checking for C compiler warning flags... -Wall -W
checking whether ln -s works... yes
checking for cmake... /opt/local/bin/cmake
checking if LD -Wl,--version-script works... yes
checking how to print strings... printf
checking for a sed that does not truncate output... /opt/local/bin/gsed
checking for grep that handles long lines and -e... /opt/local/bin/ggrep
checking for egrep... /opt/local/bin/ggrep -E
checking for fgrep... /opt/local/bin/ggrep -F
checking for ld used by /usr/bin/clang... /usr/bin/riscv64-pine64-linux-ld
checking if the linker (/usr/bin/riscv64-pine64-linux-ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... /usr/bin/llvm-nm
checking the name lister (/usr/bin/llvm-nm) interface... BSD nm
checking the maximum length of command line arguments... 1572864
checking how to convert riscv64-unknown-linux-gnu file names to riscv64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert riscv64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for /usr/bin/riscv64-pine64-linux-ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... /bin/echo
checking command to parse /usr/bin/llvm-nm output from /usr/bin/clang object... ok
checking for sysroot... no
checking for a working dd... /bin/dd
checking how to truncate binary pipes... /bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking how to run the C preprocessor... /usr/bin/clang -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if /usr/bin/clang supports -fno-rtti -fno-exceptions... yes
checking for /usr/bin/clang option to produce PIC... -fPIC -DPIC
checking if /usr/bin/clang PIC flag -fPIC -DPIC works... yes
checking if /usr/bin/clang static flag -static works... no
checking if /usr/bin/clang supports -c -o file.o... yes
checking if /usr/bin/clang supports -c -o file.o... (cached) yes
checking whether the /usr/bin/clang linker (/usr/bin/riscv64-pine64-linux-ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... yes
checking whether we are using the GNU C++ compiler... yes
checking whether /usr/bin/clang++ accepts -g... yes
checking dependency style of /usr/bin/clang++... gcc3
checking how to run the C++ preprocessor... /usr/bin/clang++ -E
checking for ld used by /usr/bin/clang++... /usr/bin/riscv64-pine64-linux-ld
checking if the linker (/usr/bin/riscv64-pine64-linux-ld) is GNU ld... yes
checking whether the /usr/bin/clang++ linker (/usr/bin/riscv64-pine64-linux-ld) supports shared libraries... yes
/opt/local/bin/ggrep: warning: stray \ before -
checking for /usr/bin/clang++ option to produce PIC... -fPIC -DPIC
checking if /usr/bin/clang++ PIC flag -fPIC -DPIC works... yes
checking if /usr/bin/clang++ static flag -static works... no
checking if /usr/bin/clang++ supports -c -o file.o... yes
checking if /usr/bin/clang++ supports -c -o file.o... (cached) yes
checking whether the /usr/bin/clang++ linker (/usr/bin/riscv64-pine64-linux-ld) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether make supports nested variables... (cached) yes
checking for sin in -lm... yes
checking assert.h usability... yes
checking assert.h presence... yes
checking for assert.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking io.h usability... no
checking io.h presence... no
checking for io.h... no
checking search.h usability... yes
checking search.h presence... yes
checking for search.h... yes
checking for unistd.h... (cached) yes
checking for an ANSI C-conforming const... yes
checking whether byte ordering is bigendian... no
checking for off_t... yes
checking for size_t... yes
checking whether time.h and sys/time.h may both be included... yes
checking whether struct tm is in sys/time.h or time.h... time.h
checking for _LARGEFILE_SOURCE value needed for large files... no
checking whether optarg is declared... yes
checking size of size_t... 8
checking for signed size type... int64_t
checking for mmap... yes
checking for setmode... no
checking for getopt... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for inflateEnd in -lz... yes
checking zlib.h usability... yes
checking zlib.h presence... yes
checking for zlib.h... yes
checking for libdeflate_zlib_decompress in -ldeflate... no
configure: error: libdeflate library not found at /opt/local/lib
Command failed:  cd "/opt/local/var/macports/build/_home_svacchanda_macstrop_graphics_tiff/tiff/work/tiff-4.6.0" && ./configure --prefix=/opt/local --disable-jbig --disable-webp --with-x=yes --with-docdir=/opt/local/share/doc/tiff --with-jpeg-include-dir=/opt/local/include --with-jpeg-lib-dir=/opt/local/lib --with-lerc-include-dir=/opt/local/include --with-lerc-lib-dir=/opt/local/lib --with-libdeflate-include-dir=/opt/local/include --with-libdeflate-lib-dir=/opt/local/lib --with-lzma-include-dir=/opt/local/include --with-lzma-lib-dir=/opt/local/lib --with-zlib-include-dir=/opt/local/include --with-zlib-lib-dir=/opt/local/lib --with-zstd-include-dir=/opt/local/include --with-zstd-lib-dir=/opt/local/lib 
Exit code: 1
Error: Failed to configure tiff: consult /opt/local/var/macports/build/_home_svacchanda_macstrop_graphics_tiff/tiff/work/tiff-4.6.0/config.log
Error: Failed to configure tiff: configure failure: command execution failed
Error: See /opt/local/var/macports/logs/_home_svacchanda_macstrop_graphics_tiff/tiff/main.log for details.
```

```
sh-5.1# port -v installed libdeflate
The following ports are currently installed:
  libdeflate @1.21_0 (active) requested_variants='' platform='linux 3' archs='riscv64' date='2024-09-15T11:11:20+0000'
sh-5.1# port contents libdeflate
Port libdeflate @1.21_0 contains:
  /opt/local/bin/libdeflate-gunzip
  /opt/local/bin/libdeflate-gzip
  /opt/local/include/libdeflate.h
  /opt/local/lib64/cmake/libdeflate/libdeflate-config-version.cmake
  /opt/local/lib64/cmake/libdeflate/libdeflate-config.cmake
  /opt/local/lib64/cmake/libdeflate/libdeflate-targets-macports.cmake
  /opt/local/lib64/cmake/libdeflate/libdeflate-targets.cmake
  /opt/local/lib64/libdeflate.a
  /opt/local/lib64/libdeflate.so
  /opt/local/lib64/libdeflate.so.0
  /opt/local/lib64/pkgconfig/libdeflate.pc
```

**Comment by barracuda156** [comment]
@RJVB Ah, so this is what happens when we have something installing into `${prefix}/lib64` instead of the canonical one.

**Comment by barracuda156** [comment]
`libdeflate` CMakeLists do not actually hardcode anything, they simply use `CMAKE_INSTALL_LIBDIR`. So this looks like an issue on our side to fix.

**Comment by RJVB** [comment]
>`libdeflate` CMakeLists do not actually hardcode anything, they simply use `CMAKE_INSTALL_LIBDIR`. So this looks like an issue on our side to fix.

The question then becomes why that variable is set the way it is... I'm open to proposals how to address that (I don't have a port:libdeflate and I'd prefer not to add one just for this).

Meanwhile, what do

> pkg-config --cflags-only-I libdeflate

and

> pkg-config --variable=libdir  libdeflate

tell you?

**Comment by barracuda156** [comment]
Same issue with `libjpeg-turbo` by the way (which is Macstrop port). Also installs libs into `${prefix}/lib64`.

For `libdeflate` I have fixed it by passing `-DCMAKE_INSTALL_LIBDIR=${prefix}/lib`. This can be submitted to upstream MacPorts, I think. Nothing specifically linuxy.

**Comment by barracuda156** [comment]
I have already rebuilt libdeflate, but here is the output for `libjpeg-turbo`:
```
sh-5.1# pkg-config --variable=libdir  libjpeg
Package libjpeg was not found in the pkg-config search path.
Perhaps you should add the directory containing `libjpeg.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libjpeg' found
sh-5.1# pkg-config --cflags-only-I libjpeg
Package libjpeg was not found in the pkg-config search path.
Perhaps you should add the directory containing `libjpeg.pc'
to the PKG_CONFIG_PATH environment variable
No package 'libjpeg' found
```
I.e. to no surprise nothing is found, because pkgconfig files are also in lib64.

I have checked the same commands on macOS, and the work correctly, returning what is expected.

**Comment by barracuda156** [comment]
@RJVB I can confirm that the same solution works for `libjpeg-turbo`. I could submit a PR for that one, of course, but perhaps this should be rather fixed in cmake PG?

**Comment by RJVB** [comment]
René J.V. Bertin wrote on 20240915::13:40:16 re: "Re: [RJVB/macstrop] tiff fails to install on Linux: `configure: error: libdeflate library not found at /opt/local/lib` (Issue #118)"

I repeat:

>>`libdeflate` CMakeLists do not actually hardcode anything, they simply use `CMAKE_INSTALL_LIBDIR`. So this looks like an issue on our side to fix.
>
>The question then becomes why that variable is set the way it is... I'm open to proposals how to address that (I don't have a port:libdeflate and I'd prefer not to add one just for this).

That would be the proper and AFAIAC only way to address this issue. Either as a port:cmake variant to force CMAKE_INSTALL_LIBDIR to `lib` ($prefix/lib) or else as a simple patch to prevent it from using `lib64` ($prefix/lib64). BTW, the variable appears not to be set by default but is set in cmake include file like GNUInstallDirs.cmake or ECM's KDEInstallDirs.cmake .

An alternative could have been to add `-DCMAKE_INSTALL_LIBDIR` to the cmake-1.1 PG , but that's something I should have done as soon as I discovered ports installing into $prefix/lib/$arch-linux-gnu . That would have been a lot cleaner but I've decided to live with that design for too long now.
